### DISCUSSION TITLE
pool.Exec treats all args as text?

### DISCUSSION THREAD

**Discussion by konart** [jackc/pgx#1486] [Q&A]
I'm getting SQL errors for the following query:

```
	putQuery = `insert into item (hash, item_id, candidate_id, source_id, error, success, body, attempt)
		    select v.hash, v.item_id, v.candidate_id, s.id, v.error, v.success, v.body, v.attempt
		      from (
			    values ($1, $2, $3, $4, $5, $6, $7, $8)
			   ) as v
			   (hash, item_id, candidate_id, source, error, success, body, attempt)
		      join source s on s.name = v.source`
```

`success`, and `attempt` fields are boolean and integer respectfully. The arguments passed to `Exec()` function are bool and an int, called like this:

```
	_, err := r.db.Exec(ctx, putQuery,
		result.Hash
		result.ItemID,
		result.CandidateID,
		result.Source,
		result.Error,
		result.Success,
		result.Body,
		result.Attempt)

// result
type Result struct {
	ItemID      string
	CandidateID string
	Error       string
	Success     bool
	Body        string
	Attempt     int
	Hash        string
	Source      InputSource
}
```

Everything looks ok to me, yet postgres begs to disagree:

`inserting result error: ERROR: column \"attempt\" is of type integer but expression is of type text (SQLSTATE 42804)`

**Comment by jackc** [comment]
I suspect that PostgreSQL has no way to determine the expected types of the bind placeholders in the values clause. Try specifying the type in your query. e.g. `$8::int`.
